### DISCUSSION TITLE
NullConstraintViolationException when a call is made to OCAPI endpoint /customers/auth for JWT

### DISCUSSION THREAD

**Discussion by redz0701** [SalesforceCommerceCloud/sfcc-ci#440] [Team Posts]
Hi All,
I am getting **NullConstraintViolationException** when trying to get JWT using 
**POST https://hostname:port/dw/shop/vx_x/customers/auth?client_id=xxxx**
Also passing type: guest in the header.

Error:
 "type": "NullConstraintViolationException",
"message": "The null value constraint for parameter 'Body' was violated. Null isn't allowed. Please provide a value."

Any help is appreciated.

Thanks in Advance.

**Comment by rajanand616** [comment]
Hi Rohit,

I have tried the Customer Oauth call by adding the type: guest in the
header and I am getting the same error as follows
{
    "_v": "18.8",
    "fault": {
        "arguments": {
            "parameter": "Body"
        },
        "type": "NullConstraintViolationException",
        "message": "The null value constraint for parameter 'Body' was
violated. Null isn't allowed. Please provide a value."
    }
}

So, to get the JWT token you need to add the *type: guest* in the body of
the request.
Please try it and let me know.

Request Body as follows:

{
  "type" : "guest"
}

Regards,
Anand

**Comment by redz0701** [comment]
Thanks, Anand. It worked.
